File: app/src/main/java/com/kevin/java/annotation/event/OnClick.java
package com.kevin.java.annotation.event;

import android.view.View;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@EventBus(eventType = View.OnClickListener.class, eventMethod = "setOnClickListener")
public @interface OnClick {

    int[] value();
}

File: app/src/main/java/com/kevin/injecttest/annotation/event/OnClick.kt
package com.kevin.injecttest.annotation.event

import android.view.View


@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Event("setOnClickListener", View.OnClickListener::class, "onClick")
annotation class OnClick(vararg val ids: Int)